BlackCubes/learning-typescript
# Eg 1 typescript starting point

## Changes
1. Added gitignore file with some ignores
2. Installed npm.
3. Installed typescript as a dev-dependency.
4. Created a tsconfig.json file.
5. Added a script to compile typescript in package.json.
6. Create an index.html.
7. Added a TS version of the JS functionalities in using-ts.ts.
8. Compile typescript from using-ts.ts to using-ts.js.

## Purpose
Need to learn how to use the basics of TypeScript.

## Learning
The approach is to go through and learn **Section 1: Getting Started** of *Understanding TypeScript - 2021 Edition* by Maximilian Schwarzmüller. The other approach is to read some of the TypeScript official documentation, and learn how to compile TypeScript from reading *Compile TypeScript with npm* by James Edwards.

[Understanding TypeScript - 2021 Edition](https://www.udemy.com/course/understanding-typescript/).
[TypeScript documentation](https://www.typescriptlang.org/docs/).
[Compile TypeScript with npm](https://www.devextent.com/npm-compile-typescript/).

## Testing
1. Pull in the changes to your local copy of this branch.
2. Go to the `getting-started` folder.
3. Run `npm install` to install TypeScript as a dev-dependency (and if you wish to try it yourself).
4. Use liveserver or double-click on `index.html` to start the application.
5. Ensure there are no errors on the browser's console.

Closes #1
